### PR TITLE
refactor: updating `vpn_gateway` to use `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR updates the Data Source and Resource `azurerm_vpn_gateway` to use `hashicorp/go-azure-sdk`

### CODE REVIEW

File: internal/services/network/vpn_gateway_resource.go [hunk 1]
```diff
@@ -196,130 +195,120 @@ func resourceVPNGateway() *pluginsdk.Resource {
...
+			input0 := val["instance_0_bgp_peering_address"].([]interface{})
+			input1 := val["instance_1_bgp_peering_address"].([]interface{})
 
-			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
-				return fmt.Errorf("waiting for creation/update of %q: %+v", id, err)
-			}
-			if err := waitForCompletion(d, ctx, client, id.ResourceGroup, id.Name); err != nil {
-				return err
+			if len(input0) > 0 || len(input1) > 0 {
+				if len(input0) > 0 {
```
Comment (mbfrahry) [inline internal/services/network/vpn_gateway_resource.go:261]:
```suggestion
				if len(input0) > 0 && input0[0] != nil {
```
Potential crash

File: internal/services/network/vpn_gateway_resource.go [hunk 2]
```diff
@@ -196,130 +195,120 @@ func resourceVPNGateway() *pluginsdk.Resource {
...
+					val := input0[0].(map[string]interface{})
+					(*props.BgpSettings.BgpPeeringAddresses)[0].CustomBgpIPAddresses = utils.ExpandStringSlice(val["custom_ips"].(*pluginsdk.Set).List())
+				}
+				if len(input1) > 0 {
```
Comment (mbfrahry) [inline internal/services/network/vpn_gateway_resource.go:265]:
```suggestion
				if len(input1) > 0 && input1[0] != nil {
```

File: internal/services/network/vpn_gateway_resource.go [hunk 3]
```diff
@@ -329,45 +318,37 @@ func resourceVPNGatewayUpdate(d *pluginsdk.ResourceData, meta interface{}) error
 		if d.HasChange("bgp_settings.0.instance_0_bgp_peering_address") {
 			if input := val["instance_0_bgp_peering_address"].([]interface{}); len(input) > 0 {
 				val := input[0].(map[string]interface{})
-				(*existing.VpnGatewayProperties.BgpSettings.BgpPeeringAddresses)[0].CustomBgpIPAddresses = utils.ExpandStringSlice(val["custom_ips"].(*pluginsdk.Set).List())
+				(*model.Properties.BgpSettings.BgpPeeringAddresses)[0].CustomBgpIPAddresses = utils.ExpandStringSlice(val["custom_ips"].(*pluginsdk.Set).List())
```
Comment (mbfrahry) [inline internal/services/network/vpn_gateway_resource.go:321]: Potential crash

File: internal/services/network/vpn_gateway_resource.go [hunk 4]
```diff
@@ -329,45 +318,37 @@ func resourceVPNGatewayUpdate(d *pluginsdk.ResourceData, meta interface{}) error
...
 			}
 		}
 		if d.HasChange("bgp_settings.0.instance_1_bgp_peering_address") {
 			if input := val["instance_1_bgp_peering_address"].([]interface{}); len(input) > 0 {
 				val := input[0].(map[string]interface{})
-				(*existing.VpnGatewayProperties.BgpSettings.BgpPeeringAddresses)[1].CustomBgpIPAddresses = utils.ExpandStringSlice(val["custom_ips"].(*pluginsdk.Set).List())
+				(*model.Properties.BgpSettings.BgpPeeringAddresses)[1].CustomBgpIPAddresses = utils.ExpandStringSlice(val["custom_ips"].(*pluginsdk.Set).List())
```
Comment (mbfrahry) [inline internal/services/network/vpn_gateway_resource.go:327]: And here

**Review by mbfrahry — APPROVED** [review]
LGTM but we should fix up some of the potential crashes before merging

**Comment by stephybun** [comment]
Applied review comments and tests look good.

**Pull request merged**